### ISSUE TITLE
Connecting to Haxball through a different port?

### ISSUE THREAD

**Issue by LGosh** [haxball/haxball-issues#700]
Hi, I'm in the college right now where they limit out connection. I'm not entirely sure if they block ports of certain stuff, or what method do they use to limit it. However, I can't play Haxball because it says "Failed" after trying to join any room. However, I can play Agar.io without a problem. This is weird isn't it? Since as far as I understand both of them require RTC connection? 

VPN helps me bypass this limitation and play Haxball but with excessively huge pings, and thus it's not playable. 

Do you guys think there can be another alternative for this limitation? For example change the port that Chrome is using? Or something, I don't know, and sorry if I wrote bulshit because I'm not a pro at this.

EDIT: Not a duplicate of #236 because they asked about a workaround from the Haxball side, whereas I'm asking for a workaround on my side.

**Comment by basro** [comment]
I believe Agar.io uses WebSockets while HaxBall uses WebRTC. So unfortunately it's not strange for agar.io to work while webrtc connections fail.

WebSockets are very unlikely to get blocked by firewalls since they often use TCP port 80 which is the same as http uses, firewalls that allow http will almost always allow websockets.

WebRTC is using UDP and uses randomly selected ports from a a big range (called ephemeral port range). It is a peer to peer technology which lets users connect to each other directly (as in, once you have connected to a room the data goes directly from your pc to that room's host pc without it going through servers controlled by me).
Being peer to peer is good for latency and also makes the whole thing affordable (I'd not be able to pay for the amount of servers needed to host all the rooms). But the downside is that connections can fail more often because of firewalls or routers that simply refuse to play nice.

The problem and the solution is often in the router, but since you are in a college campus it's unlikely that you'll be able to change things in that end.

**Comment by LGosh** [comment]
Thank you for your quick, detailed, and informative response. So I take it's either VPN and huge lags, or nothing, right?

**Comment by basro** [comment]
A VPN doesn't guarantee "huge lags". It will add some lag, because all packets will be routed through their servers first, but if you find a VPN with servers close enough to you the added latency could be as small as 30 extra milliseconds. So maybe try looking for VPN's that care about latency and have servers near you.

Other than that, yes, I'm sorry but nothing.

**Comment by LGosh** [comment]
Ok, thanks, I appreciate your detailed answer.
